tests/python/quantization/test_quantization.py

```diff
@@ -673,8 +673,9 @@ def test_optimal_threshold_adversarial_case():
     # The worst case for the optimal_threshold function is when the values are concentrated
     # at one edge: [0, 0, ..., 1000]. (histogram)
     # We want to make sure that the optimal threshold in this case is the max.
-    arr = np.array([2]*1000)
-    res = mx.contrib.quant._get_optimal_threshold(arr, num_quantized_bins=5)
+    arr = np.array([2] * 1000)
+    for dtype in ['uint8', 'int8', 'auto']:
+        res = mx.contrib.quant._get_optimal_threshold(arr, dtype, num_quantized_bins=5)
     # The threshold should be 2.
     assert res[3] - 2 < 1e-5
```

Comment:
shouldnt this assert be for each of the dtypes ?

Reply:
For uint8, it should have more accuracy, thus the threshold should be half. I will edit here.

Reply:
For this case, the ideal result should be `2` no matter what dtype is. So we don't need to use different assert here. @reminisce Do you have other comments here?

Reply:
can we move the assert inside the for loop in this case ?

Reply:
Done.